### ISSUE TITLE
pre-training model

### ISSUE THREAD

**Issue by nhatnxn** [V-Sense/ACTION-Net#1]
Do you have any pre-training models or check-point? 
Thanks

**Comment by villawang** [comment]
Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.

**Comment by villawang** [comment]
> Do you have any pre-training models or check-point?
> Thanks

Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.

**Comment by nhatnxn** [comment]
> > Do you have any pre-training models or check-point?
> > Thanks
> 
> Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.

**Comment by nhatnxn** [comment]
> > Bạn có bất kỳ mô hình đào tạo trước hoặc điểm kiểm tra nào không? 
> > Cảm ơn
> 
> Xin chào, chúng tôi đã đổi tên và xây dựng lại mô hình cho bản phát hành công khai. Vì vậy, các trạm kiểm soát được đào tạo trước đó sẽ không hoạt động cho việc này. Nếu bạn tuân theo cài đặt đào tạo trong bài báo của chúng tôi, bạn sẽ nhận được kết quả tương tự.
thanks. I hope it will be coming soon

**Comment by nhatnxn** [comment]
Thanks. I hope it will be coming soon

**Comment by kinfeparty** [comment]
> > Do you have any pre-training models or check-point?
> > Thanks
> 
> Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.

Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
Can you give me some advice to get the results in papaer with the small batch size?

**Comment by villawang** [comment]
> > > Do you have any pre-training models or check-point?
> > > Thanks
> > 
> > 
> > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> 
> Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> Can you give me some advice to get the results in papaer with the small batch size?

Hi there,

Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
```
python3 train.py --is_train --is_shift \
                 --dataset EgoGesture --clip_len 8 \
                 --shift_div 8 --wd 1e-5 --dropout 0 \
                 --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
                 --lr 5e-3 --base_model resnet50 --epochs 25
```

**Comment by kinfeparty** [comment]
> > > > Do you have any pre-training models or check-point?
> > > > Thanks
> > > 
> > > 
> > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > 
> > 
> > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > Can you give me some advice to get the results in papaer with the small batch size?
> 
> Hi there,
> 
> Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> 
> ```
> python3 train.py --is_train --is_shift \
>                  --dataset EgoGesture --clip_len 8 \
>                  --shift_div 8 --wd 1e-5 --dropout 0 \
>                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
>                  --lr 5e-3 --base_model resnet50 --epochs 25
> ```

I will try these parameters setting, thanks for your answer.

**Comment by kinfeparty** [comment]
> > > > Do you have any pre-training models or check-point?
> > > > Thanks
> > > 
> > > 
> > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > 
> > 
> > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > Can you give me some advice to get the results in papaer with the small batch size?
> 
> Hi there,
> 
> Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> 
> ```
> python3 train.py --is_train --is_shift \
>                  --dataset EgoGesture --clip_len 8 \
>                  --shift_div 8 --wd 1e-5 --dropout 0 \
>                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
>                  --lr 5e-3 --base_model resnet50 --epochs 25
> ```



> > > > Do you have any pre-training models or check-point?
> > > > Thanks
> > > 
> > > 
> > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > 
> > 
> > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > Can you give me some advice to get the results in papaer with the small batch size?
> 
> Hi there,
> 
> Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> 
> ```
> python3 train.py --is_train --is_shift \
>                  --dataset EgoGesture --clip_len 8 \
>                  --shift_div 8 --wd 1e-5 --dropout 0 \
>                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
>                  --lr 5e-3 --base_model resnet50 --epochs 25
> ```

Hello, I've successfully reproduce the result in egogesture. I want to do the experiment of 16 clip_len in another dataset. But the accuracy is lower than the result of 8 clip_len. By the way, the batch size is 24 for 8 clip_len and 12 for 16 clip_len. Does the small batch size cause this error? Or other parameters needed to be changed?  If it requires higher batch size, I will rerun the experiment(the GPU memory isn't enough for now).

**Comment by villawang** [comment]
Hi there, you might consider halving the learning rate for 16 clips. A small batch size may decrease the performance as well. Actually, for the jester and EgoGesture dataset, we found 8 clips are enough to capture the action and 16 clips will give similar results both for our method and other 2D CNN approaches.

**Comment by villawang** [comment]
> > > > > Do you have any pre-training models or check-point?
> > > > > Thanks
> > > > 
> > > > 
> > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > 
> > > 
> > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > Can you give me some advice to get the results in papaer with the small batch size?
> > 
> > 
> > Hi there,
> > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > ```
> > python3 train.py --is_train --is_shift \
> >                  --dataset EgoGesture --clip_len 8 \
> >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > ```
> 
> > > > > Do you have any pre-training models or check-point?
> > > > > Thanks
> > > > 
> > > > 
> > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > 
> > > 
> > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > Can you give me some advice to get the results in papaer with the small batch size?
> > 
> > 
> > Hi there,
> > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > ```
> > python3 train.py --is_train --is_shift \
> >                  --dataset EgoGesture --clip_len 8 \
> >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > ```
> 
> Hello, I've successfully reproduce the result in egogesture. I want to do the experiment of 16 clip_len in another dataset. But the accuracy is lower than the result of 8 clip_len. By the way, the batch size is 24 for 8 clip_len and 12 for 16 clip_len. Does the small batch size cause this error? Or other parameters needed to be changed? If it requires higher batch size, I will rerun the experiment(the GPU memory isn't enough for now).

Hi there, you might consider halving the learning rate for 16 clips. A small batch size may decrease the performance as well. Actually, for the jester and EgoGesture dataset, we found 8 clips are enough to capture the action and 16 clips will give similar results both for our method and other 2D CNN approaches.

**Comment by kinfeparty** [comment]
> > > > > > Do you have any pre-training models or check-point?
> > > > > > Thanks
> > > > > 
> > > > > 
> > > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > > 
> > > > 
> > > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > > Can you give me some advice to get the results in papaer with the small batch size?
> > > 
> > > 
> > > Hi there,
> > > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > > ```
> > > python3 train.py --is_train --is_shift \
> > >                  --dataset EgoGesture --clip_len 8 \
> > >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> > >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> > >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > > ```
> > 
> > 
> > > > > > Do you have any pre-training models or check-point?
> > > > > > Thanks
> > > > > 
> > > > > 
> > > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > > 
> > > > 
> > > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > > Can you give me some advice to get the results in papaer with the small batch size?
> > > 
> > > 
> > > Hi there,
> > > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > > ```
> > > python3 train.py --is_train --is_shift \
> > >                  --dataset EgoGesture --clip_len 8 \
> > >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> > >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> > >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > > ```
> > 
> > 
> > Hello, I've successfully reproduce the result in egogesture. I want to do the experiment of 16 clip_len in another dataset. But the accuracy is lower than the result of 8 clip_len. By the way, the batch size is 24 for 8 clip_len and 12 for 16 clip_len. Does the small batch size cause this error? Or other parameters needed to be changed? If it requires higher batch size, I will rerun the experiment(the GPU memory isn't enough for now).
> 
> Hi there, you might consider halving the learning rate for 16 clips. A small batch size may decrease the performance as well. Actually, for the jester and EgoGesture dataset, we found 8 clips are enough to capture the action and 16 clips will give similar results both for our method and other 2D CNN approaches.

The halving means that no matter the batchsize is 24 or 12, I need to set 24e-4 for 16 clip_len model training?

**Comment by villawang** [comment]
> > > > > > > Do you have any pre-training models or check-point?
> > > > > > > Thanks
> > > > > > 
> > > > > > 
> > > > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > > > 
> > > > > 
> > > > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > > > Can you give me some advice to get the results in papaer with the small batch size?
> > > > 
> > > > 
> > > > Hi there,
> > > > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > > > ```
> > > > python3 train.py --is_train --is_shift \
> > > >                  --dataset EgoGesture --clip_len 8 \
> > > >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> > > >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> > > >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > > > ```
> > > 
> > > 
> > > > > > > Do you have any pre-training models or check-point?
> > > > > > > Thanks
> > > > > > 
> > > > > > 
> > > > > > Hi there, we have renamed and reconstructed the model for the public release. So previous trained checkpoints will not work for this. If you follow the training setting in our paper, you should get similar results.
> > > > > 
> > > > > 
> > > > > Hello! I only have 4 16GB GPU for training with 32batch size,and the test result is 93.6 in egogesture.
> > > > > Can you give me some advice to get the results in papaer with the small batch size?
> > > > 
> > > > 
> > > > Hi there,
> > > > Can you try the training setting below for batch size equal to 32? I can reproduce the results from the paper.
> > > > ```
> > > > python3 train.py --is_train --is_shift \
> > > >                  --dataset EgoGesture --clip_len 8 \
> > > >                  --shift_div 8 --wd 1e-5 --dropout 0 \
> > > >                  --cuda_id $cuda_id --batch_size 32 --lr_steps 10 15 20 \
> > > >                  --lr 5e-3 --base_model resnet50 --epochs 25
> > > > ```
> > > 
> > > 
> > > Hello, I've successfully reproduce the result in egogesture. I want to do the experiment of 16 clip_len in another dataset. But the accuracy is lower than the result of 8 clip_len. By the way, the batch size is 24 for 8 clip_len and 12 for 16 clip_len. Does the small batch size cause this error? Or other parameters needed to be changed? If it requires higher batch size, I will rerun the experiment(the GPU memory isn't enough for now).
> > 
> > 
> > Hi there, you might consider halving the learning rate for 16 clips. A small batch size may decrease the performance as well. Actually, for the jester and EgoGesture dataset, we found 8 clips are enough to capture the action and 16 clips will give similar results both for our method and other 2D CNN approaches.
> 
> The halving means that no matter the batchsize is 24 or 12, I need to set 24e-4 for 16 clip_len model training?

Halving your learning rate when your batch size is halved. Details can be referred [here](https://arxiv.org/pdf/1706.02677.pdf).
